mantle/kola/harness.go

```diff
@@ -804,16 +812,55 @@ func runProvidedTests(testsBank map[string]*register.Test, patterns []string, mu
 	if len(testsToRerun) > 0 && rerun {
 		newOutputDir := filepath.Join(outputDir, "rerun")
 		fmt.Printf("\n\n======== Re-running failed tests (flake detection) ========\n\n")
-		reRunErr := runProvidedTests(testsBank, testsToRerun, multiply, false, allowRerunSuccess, pltfrm, newOutputDir, propagateTestErrors)
-		if allowRerunSuccess {
+		reRunErr := runProvidedTests(testsBank, testsToRerun, multiply, false, rerunSuccessTags, pltfrm, newOutputDir, propagateTestErrors)
+
+		// Let's see which tests have failed
+		if len(rerunSuccessTags) > 0 && len(filterTaggedTests(testsToRerun, tagged)) == 0 {
```

Comment:
seems like we could just call a function here to go through to see if all re-ran tests qualified:

```
if len(rerunSuccessTags) > 0 && allTestsAllowRerunSuccess(testsToRerun, rerunSuccessTags) {
    return ReRunErr
}
```

So we'd make a new function `allTestsAllowRerunSuccess()` that handles all the logic.

Reply:
actually i was going to make more patches and completely split the logic here to add support for `rerun` and `allow-rerun-success --tags=xxx,tests=yyy:,opt1=v1,optX=vX`.
that's why this PR is as simple as possible for now

Reply:
It's still not clear to me how the current functions allow that future support to exist. Maybe we update the functions when we add the new features..

I took a stab at writing a `allTestsAllowRerunSuccess()` in https://github.com/dustymabe/coreos-assembler/commit/c466aa35838d16d922d24ed53b8798bb15657295. It's a bit easier to read to me, but I don't know if it's appropriate.

Reply:
ok, added your changes (with some minor on top) and you as co-author